### PR TITLE
V5 init

### LINKED ISSUE #324: `runScenarios` should accept `ScenarioConfiguration`
Breaking change

### LINKED ISSUE #326: Bump version to 5
In `appveyor.yml`.

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #324
Fixes #326

### CODE REVIEW

**Comment by PavelBal** [comment]
@svavil bumping version to 5 and implementing #330

File: R/error-checks.R [hunk 1]
```diff
@@ -1 +1,20 @@
 
+#' Validate parameter list structure
+#'
+#' @param parameterStructure Object to be checked. Expected is a named list
+#' with names "paths", "values", and "units".
+#'
+#' @keywords internal
+#'
+#' @return `TRUE` if validation succeeded (silently). Throws an error otherwise.
+.validateParametersStructure <- function(parameterStructure, argumentName = NULL,
```
Comment (PavelBal) [inline R/error-checks.R:10]: Moved some repetitive code to an internal function.

File: R/scenario-configuration.R [hunk 2]
```diff
@@ -146,6 +146,10 @@ a parameter sheet from the list"
     #' @field individualId Id of the individual as specified in "IndividualParameters.xlsx".
     #' If `NULL` (default), the individual as defined in the simulation file will be simulated.
     individualId = NULL,
+    #' @field outputPaths a list of output paths for which the results will be
```
Comment (PavelBal) [inline R/scenario-configuration.R:149]: Output paths are defined within scenario configuration now.

File: R/utilities-scenario-configuration.R [hunk 3]
```diff
@@ -1,68 +1,87 @@
...
+  validateIsString(scenarioNames)
+  validateIsOfType(projectConfiguration, ProjectConfiguration)
+
   # Current scenario definition structure:
   # "Scenario_name", "IndividualId", "ModelParameterSheets", "ApplicationProtocol",
   # "SimulationTime", "SimulationTimeUnit", "SteadyState", "ModelFile"
   colTypes <- c("text", "text", "text", "text", "numeric", "text", "logical", "text")
-  data <- readExcel(
+  wholeData <- readExcel(
     path = file.path(
```
Comment (PavelBal) [inline R/utilities-scenario-configuration.R:33]: `data` is used later and is a subset of `wholeData` for the specific scenario.

File: R/utilities-scenario-configuration.R [hunk 4]
```diff
@@ -1,68 +1,87 @@
...
   )
-  # Select the scenario
-  if (!scenarioConfiguration$scenarioName %in% data$Scenario_name) {
-    stop("readScenarioDefinition: Scenario '", scenarioConfiguration$scenarioName, "' is not specified!")
-  }
-  data <- data[data$Scenario_name == scenarioConfiguration$scenarioName, ]
 
-  # Parameter sheets
-  scenarioConfiguration$removeParamSheets()
-  paramSheets <- gsub(data$ModelParameterSheets, pattern = " ", replacement = "", fixed = TRUE)
```
Comment (PavelBal) [inline R/utilities-scenario-configuration.R:43]: This all is happeing later in a loop

File: R/utilities-scenarios.R [hunk 5]
```diff
@@ -16,31 +15,46 @@
 #' `SimulatioResults` objects produced by running the simulation, and output values
 #' of the `SimulationResults`.
 #' @export
-runScenarios <- function(scenarioNames, scenarioConfiguration, customParams = NULL,
+runScenarios <- function(scenarioConfigurations, customParams = NULL,
```
Comment (PavelBal) [inline R/utilities-scenarios.R:18]: Here, we have to pass scenario configurations ready to run instead of initializing them within the functin.

**Comment by PavelBal** [comment]
@svavil please review this ASAP, so I can continue working on the vignettes.

I have disabled the tests for sensitivity calculation as I just cannot understand what they are doing (e.g. comparing dimensions to some magic numbers...)  and why they are failing. This should be fixed in a separate PR.

**Review by svavil — APPROVED** [review]

**Pull request merged**